src/dspbase.jl

```diff
@@ -718,7 +718,8 @@ function conv(u::AbstractArray{<:BLAS.BlasFloat, N},
     conv(u, float(v))
 end
 
-function conv(A::AbstractArray, B::AbstractArray)
+function conv(A::AbstractArray{<:Number, N},
+              B::AbstractArray{<:Number, P}) where {N, P}
     maxnd = max(ndims(A), ndims(B))
     return conv(cat(A, dims=maxnd), cat(B, dims=maxnd))
```

Comment:
You can use the static parameters here to find maxnd: `maxnd = max(N, P)`

Also definitely picking at nits but why `P`? `M` and `N` would be more common choices for the number of dimensions.